src/Native/System.Security.Cryptography.Native/pal_err.cpp

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_err.h"
+
+#include <openssl/err.h>
+
+extern "C" uint64_t ErrGetError()
+{
+    return ERR_get_error();
+}
+
+extern "C" void ErrErrorStringN(uint64_t e, char* buf, int32_t len)
+{
+    ERR_error_string_n(e, buf, static_cast<size_t>(len));
```

Comment:
Use pal_utilities.h and UnsignedCast(len). Will assert on < 0 and will let the > 0 -> size_t conversion be implicit in the (exceedingly unlikely. I admit) case that positive int32_t is not a valid size_t. Or just using uint32_t len but that will just move casts to managed code.